Mlem/App/Views/Shared/MarkdownWithLinkList.swift

```diff
@@ -38,12 +41,45 @@ struct MarkdownWithLinkList: View {
         VStack(spacing: Constants.main.standardSpacing) {
             Markdown(blocks, configuration: shouldBlur ? .defaultBlurred(palette: palette) : .default(palette: palette))
             if tappableLinksDisplayMode != .disabled {
-                ForEach(
-                    Array(blocks.links.filter { !$0.insideSpoiler }.enumerated()),
-                    id: \.offset
-                ) { _, link in
+                linksView(blocks.links.filter { !$0.insideSpoiler })
+            }
+        }
+    }
+    
+    @ViewBuilder
+    func linksView(_ linksData: [LinkData]) -> some View {
+        if linksData.count > 3 {
+            ForEach(Array(linksData[0..<3].enumerated()), id: \.offset) { _, link in
+                linkView(link)
+            }
+            
+            if linksCollapsed {
+                Button {
+                    withAnimation {
+                        linksCollapsed = false
+                    }
+                } label: {
+                    FooterLinkView(title: "Show \(linksData.count - 3) more links", subtitle: nil)
```

Comment:
Needs to be localized. Also I think `"n more links..."` might be better than `"Show n more links"`?

```suggestion
                    FooterLinkView(title: String(localized: "\(linksData.count - 3) more links..."), subtitle: nil)
```